### PR TITLE
Backport PMIP fix issue with mono_dont_free_domains  (#708)

### PR DESCRIPTION
**Pull request by mderoy**
* Fix issue with Pacman and domains

* Write version line

* remove unnecesary strdup

* Rename plugin

### CODE REVIEW

File: mono/mini/mixed_callstack_plugin.c [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+#if !defined(DISABLE_JIT) && defined(HOST_WIN32)
+
+static gboolean enabled;
+static mono_mutex_t mutex;
+static HANDLE fileHandle;
+PMIPProfiler pmipProfiler;
+int pmipFileNum;
+
+#define pmip_my_callstack_lock() mono_os_mutex_lock (&mutex)
```
Comment (joncham) [inline mono/mini/mixed_callstack_plugin.c]: Not a big deal, but why not change all naming if renaming?

File: msvc/libmono-static.vcxproj.filters [hunk 2]
```diff
@@ -237,7 +237,7 @@
     <ClInclude Include="..\mono\mini\mini-windows.h">
       <Filter>Header Files</Filter>
     </ClInclude>
-    <ClInclude Include="..\mono\mini\pmip_my_callstack.h">
+    <ClInclude Include="..\mono\mini\mixed_callstack_plugin.h">
```
Comment (joncham) [inline msvc/libmono-static.vcxproj.filters]: The c file should be in here too

File: mono/mini/mixed_callstack_plugin.c (hunk 1 above)
Comment (mderoy) [reply mono/mini/mixed_callstack_plugin.c]: whoops.  missed one :/

File: mono/mini/mixed_callstack_plugin.c [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+{
+	char* frame;
+	long bytesWritten = 0;
+
+	if (!enabled)
+		return;
+
+	mixed_callstack_plugin_lock ();
+	frame = g_strdup_printf ("%p;%p;%s\n", info->code, ((char*)info->code) + info->code_size, info->name ? info->name : "");
+	WriteFile(fileHandle, frame, strlen(frame), bytesWritten, NULL);
```
Comment (joncham) [inline mono/mini/mixed_callstack_plugin.c]: This should be `&bytesWritten`

Also, you could probably factor out the writing both here and in `mixed_callstack_plugin_save_method_info` above.

Comment (mderoy) [reply mono/mini/mixed_callstack_plugin.c]: going to fix the bytesWritten, but I don't think theres much value in factoring this out right now

File: msvc/libmono-static.vcxproj.filters (hunk 2 above)
Comment (mderoy) [reply msvc/libmono-static.vcxproj.filters]: It is now :)

**Pull request merged**